### DISCUSSION TITLE
Annotation simplification

### DISCUSSION THREAD

**Discussion by PorpoiseP** [rubberduck-vba/Rubberduck#6172] [Ideas]
This may be more of a dumb question than an actual reccomendation... 

But why not use @Description for variables and or modules as well as subroutine/functions? 

Also 

Can we get @Parameter annotations to provide variable definitions?
example
`
'@Description("Generates an unhandled error if 0.")
'@Parameter("Value to freak out over")
Sub Foo(PleaseDontBeZero As Long)

   Dim bar as single
   
   bar = 1 / PleaseDontBeZero

End Sub   
`

**Comment by retailcoder** [comment]
Not dumb, that's how we wanted it to begin with... but then edge cases took it down.

There is no way to differentiate an annotation that's for the module vs one that's for the first procedure of that module, or one that's for a variable in the declarations section; that's why they - annoyingly - all need to be different types of annotations; because resolving their respective target isn't always as straightforward as it seems... not unless we started forcing annotations to appear on top of options, but then we cannot assume options even exist in the first place.

Parameter annotations would be nice to have in RD3, especially since we get to make our own parameter info tooltips... not quite as useful in RD2 though, although we could still be showing the parameter descriptions in the CommandBar when one is selected. IIRC there's no way to natively provide actual docstrings for parameters, so these would have to work a bit differently than the other description annotations (because there's no synchronization to do).

**Reply by PorpoiseP** [reply]
Thank you for the prompt and informative response!
